### DISCUSSION TITLE
Should we have a way to stake native token?

### DISCUSSION THREAD

**Discussion by alexaunter** [galactic3/ft-lockup-contract#5] [Q&A] [answered]
Should we have a way to stake native token?

Current implementation allows staking your native tokens, to increase amount of NEAR that you'll receive upon lockup.

New draft implementation only works for FT, and if we use wrapped near - then we're unable to stake it out of the box.

Should we do anything about it on the FT lockup contract side?

**Comment by AngelBlock** [comment]
I'd like to advice that ability to stake the lockup is an incentive and some deals may fail should end-user have no ways of staking their locked tokens. Should we discuss / consider to add native NEAR token to the lockup contract and staking functions?

Is there a plan to introduce FT staking in the near future? Can wrapped near be staked?

**Comment by AngelBlock** [comment]
I was wondering if stNEAR can be used for that purpose, is there a documentation on the token implementation?

**Comment by frol** [comment, marked as answer]
This discussion can be more generic: Should the locked tokens participate in economy? There are a few concerns I have already heard and observed:

1. Locked tokens create inequality between those who invested at the current token price vs those who got allocated tokens. If, for example, the locked tokens can be used as voting power, this immediately creates disbalance in the system.
2. Termination and regular withdrawal process gets immediately more complicated.

The stakeholders I reached out to (Aurora, Pagoda, and EK) do not have the need for this scenario as of now. We will need to hear the perspective from NF side regarding this.
